src/hooks/useFetch.ts

```diff
@@ -31,6 +31,28 @@ export function getRequestUrl({
   return urlObj.toString();
 }
 
+// 커스텀 에러 클래스
+export class FetchError extends Error {
+  status: number;
+  response: Record<string, unknown>;
```

Comment:
[제안] 제네릭으로 확장가능하도록 구현하면 재사용성이 좋아지겠네요

```suggestion
export class FetchError<T> extends Error {
  readonly status: number;
  readonly response: T;
```